nbformat/validator.py

```diff
@@ -294,7 +327,8 @@ def iter_validate(nbdict=None, ref=None, version=None, version_minor=None,
     if version is None:
         version, version_minor = get_version(nbdict)
 
-    validator = get_validator(version, version_minor, relax_add_props=relax_add_props)
+    validator = get_validator(version, version_minor, relax_add_props=relax_add_props,
+                              use_fast=False)
```

Comment:
Hi @echarles, thanks for the feedback. So:

>use_fast is always set to False. Anyway that a consumer of the nbformat library would use it with use_fast set to True? I mean, it sounds to me reading the changes that it will not be possible for a consumer to benefit from this enhancement as it would always be disabled.

Yes, this is because there is some nested logic that required the "switch"

`validate -> iter_validate -> better_validation_errors -> validate ....`

I am looking into decoupling it so the `use_fast` is no longer needed.

>A consumer can can call validate(nb, use_fast=True), but what about the double validation question we have identified in jupyter_server? The validation is always triggered on load in nbformat?

The `read` method which also validates (but does not raise errors) uses now `use_fast=True`. Since we use validate on jupyter_server, we could add the  `use_fas=True`  there, so we would still have double validation, but now both are fast.

---

Still, I would prefer no to have the `use_fast`  parameter, so I need to think of a better way to decouple the
`validate -> iter_validate -> better_validation_errors -> validate ....` chain